### ISSUE TITLE
Proposal: Prefer dot notation for accessing object properties

### ISSUE THREAD

**Issue by davidwarrington** [davidwarrington/liquid-community-standards#2]
When accessing object properties one should always use dot notation (`product.variants`) if possible, as opposed to bracket notation (`product['variants']`).

By using dot notation we make it clear that the property being accessed is not meant to change under any circumstances. Using bracket notation increases the risk of another developer incorrectly assuming a mistake has been made, and that the property was meant to be accessed with a dynamic key (e.g `product[variants]`). With dot notation the original intentions are clearer because dynamic keys cannot be used. Dot notation also has less visual noise.

## Examples

```
{{- product.variants | json -}}

{%- comment -%}
	Accessing a property using bracket
	notation is fine if it's necessary
{%- endcomment -%}
{%- assign account_type = customer.metafields.account.type -%}
{%- assign banner_text = shop.metafields[account_type].banner_text -%}
```

**Comment by leighs-hammer** [comment]
One minor point on this, pertaining to the example I know many metafield sets use the Key for display. 

For example: 

Namespace: specifications
Key: Wheel Radius
Value: 116mm

In the above scenario the key includes a space this is fine when used programatically but would require using the [index] syntax to allow for escape chrs. :) 

Both are pretty readable as a standard I agree dot notation as preferred, key as required.

**Comment by davidwarrington** [comment]
I agree with what you're saying there, and I think that's fine to add to the example. I do think the comment "it's fine if it's necessary" already covers cases like that though? Perhaps it could benefit from clarification that the documented example isn't the only possible case where the rule would be broken? I was actually a bit torn when writing the example. Feels odd saying "you should do things this way" only to have more code documenting an exception to the rule.

I would also hope that anybody following a code styleguide like this would consider it as guidelines rather than strict rules? Seems to me that there's probably always at least one edge case for breaking a rule. Usually if it's something that could cause confusion later down the line I'd suggest adding a comment to explain why it's been done. That goes for any language
